### DISCUSSION TITLE
Custom InterfaceKernel for permeation flux

### DISCUSSION THREAD

**Discussion by lanzam213** [idaholab/moose#20044] [Q&A General] [answered]
My question relates to constructing a DGresidual and Jacobian for a custom interface kernel. I developed the interface kernel to satisfy a permeation flux model at a gas-solid interface. A summary of the problem approach and interface kernel is included below. Currently I'm stuck after looking at examples of interface kernels with DGresidual, including conjugateHeatTransfer, and attempting to solve this problem.  The solution converges but does not appear correct. 

**permeation_example.i Input** 
[permeation_example.txt](https://github.com/idaholab/moose/files/7866021/permeation_example.txt)

**Problem summary**
[Permeation_approach.pptx](https://github.com/idaholab/moose/files/7866009/Permeation_approach.pptx)

**PermeationFLuxKernel.h**
[PermeationFluxKernel.h.txt](https://github.com/idaholab/moose/files/7866078/PermeationFluxKernel.h.txt)

-Mark

**Comment by GiudGiud** [comment, marked as answer]
Hello

If it converges but the result is incorrect, then the formula for the residual must be wrong.
If you replace the interface by boundary conditions, do you get acceptable results?
If so then there must be an issue with the interface kernel. What kind of fluxes do you have across the interface? Just diffusion? Looks like there is a constant term too?

Guillaume
